### PR TITLE
Add set & get dedicated node extra fees in chain & client

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

- Added setExtraFees & getDedicatedNodePrice
- Added extraFees script
- removed unnecessary awaits

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/499

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/tfchain_client/src/contracts.ts [hunk 1]
```diff
@@ -294,23 +298,35 @@ class Contracts extends QueryContracts {
...
+  async setExtraFees(options: SetDedicatedNodeExtraFeesOptions) {
+    const extrinsic = this.client.api.tx.smartContractModule.setDedicatedNodeExtraFee(options.nodeId, options.extraFee);
+    return this.client.patchExtrinsic<RentContract>(extrinsic);
+  }
+
+  @checkConnection
+  async getDedicatedNodePrice(options: GetDedicatedNodePriceOptions) {
+    const res = await this.client.api.query.smartContractModule.dedicatedNodesExtraFee(options.nodeId);
+    return res.toPrimitive() as number;
+  }
```
Comment (AhmedHanafy725) [inline packages/tfchain_client/src/contracts.ts]: the query methods should be in the query class at the top of this module

File: packages/tfchain_client/src/nodes.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ export interface Node {
   virtualized: boolean;
   serialNumber: string;
   connectionPrice: number;
+  hasGpu: number;
```
Comment (AhmedHanafy725) [inline packages/tfchain_client/src/nodes.ts:24]: this does not exist on the node object on the chain, no?

File: packages/tfchain_client/src/nodes.ts [hunk 3]
```diff
@@ -44,6 +45,7 @@ interface NodeResources {
   sru: number;
   cru: number;
   mru: number;
+  gpu: number;
```
Comment (AhmedHanafy725) [inline packages/tfchain_client/src/nodes.ts:48]: this one too doesn't exist, no?

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: packages/tfchain_client/src/contracts.ts (hunk 1 above)
Comment (zaelgohary) [reply packages/tfchain_client/src/contracts.ts]: done.

File: packages/tfchain_client/src/nodes.ts (hunk 2 above)
Comment (zaelgohary) [reply packages/tfchain_client/src/nodes.ts]: Not yet. I removed gpu changes & will implement them once added to the chain.

File: packages/tfchain_client/src/nodes.ts (hunk 3 above)
Comment (zaelgohary) [reply packages/tfchain_client/src/nodes.ts]: same here.

File: packages/tfchain_client/src/contracts.ts [hunk 4]
```diff
@@ -146,6 +146,12 @@ class QueryContracts {
     const res = await this.client.api.query.smartContractModule.serviceContracts(options.serviceId);
     return res.toPrimitive() as unknown as ServiceContract;
   }
+
+  @checkConnection
+  async getDedicatedNodeExtraFee(options: GetDedicatedNodePriceOptions) {
```
Comment (AhmedHanafy725) [inline packages/tfchain_client/src/contracts.ts:151]: please add the return type

Comment (zaelgohary) [reply packages/tfchain_client/src/contracts.ts]: Added.

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**